src/io.ml

```diff
@@ -37,8 +37,12 @@ let write fd buf =
 let connect host port =
 	let sockaddr = Unix.ADDR_INET (host, port) in
 	let fd = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
-	Unix.connect fd sockaddr;
-	fd
+	try
+		Unix.connect fd sockaddr;
+		fd
+	with e ->
+		Unix.close fd;
```

Comment:
There is a lot of duplication with https://github.com/xapi-project/xenopsd/pull/512/files#diff-8ebf9b785c8074bebb779899dfa573a6

Why can't we use this io module in xenopsd instead? What is the difference between the two modules?

Reply:
Why there is a repo `xenopsd` and another `xenops`? Is `xenops` been used anywhere?

Reply:
There is https://github.com/xapi-project/xenops that publishes xapi
```
$ opam list --depends-on xapi-xenops
# Available packages directly depending on xapi-xenops.master for 4.05.0:
xapi-rrdd          master  Performance monitoring daemon for xapi
xapi-xenops-utils  master  Create/destroy/manipulate Xen domains
-xenops and xapi-xenops-utils.
```
If there is overlapping with xenopsd, unless we have a good reason I'd reduce duplication and use xenops here as well. I'd like to hear an opinion about this from @robhoes or @jonludlam that know the history of these packages

Reply:
Either that, or move whatever is left in here that is useful (mostly `list_domains`) into xenopsd.git, and remove this repo. IMO, this one was a disaggregation step to far... @jonludlam ?

Reply:
For the two repos depends on `xapi-xenops`, I found:
1. `xapi-rrdd` could build without installing `xapi-xenops`
2. no package depens on `xapi-xenops-utils`

Does that means we could remove/archive this repo?

Reply:
Sounds like it. I think the `list_domains` binary (part of the utils) is installed as part of xenserver, so we would need to preserve and install that as part of xenopsd